src/main/native/windows/processes-jni.cc

```diff
@@ -92,6 +90,10 @@ struct NativeOutputStream {
     handle_ = INVALID_HANDLE_VALUE;
   }
 
+  void SetHandle(HANDLE handle) {
+    handle_ = handle;
```

Comment:
Do you need to release the old handle or check that the old value was invalid?

Reply:
The old code didn't do anything like that (it just accessed the member variable directly), so this is currently equivalent to the old behavior. I don't mind changing it either way, but I didn't look into this class much, so am not sure what would be the best choice.

Reply:
It shouldn't hurt to call `Close()` then. Could you please do so?

Reply:
Actually, if we call `Close()` here, this would have introduced a bug, because then `closed_` is true, even though we assign a new handle and then calls to `ReadStream` would fail, because it just does `return 0` when the stream is marked as closed.

The method is only called in two places in this file and only during initialization (so SetHandle is never called twice on the same object), so maybe let's just keep it as is? If we refactor this in the future, we could consider removing the SetHandle method completely and just pass the HANDLE to the constructor.

Reply:
SGTM.